packages/flutter/lib/src/rendering/unconstrained_box.dart

```diff
@@ -0,0 +1,170 @@
+/// on an infinite canvas with no constraints. This container will then expand
+/// as much as it can within its own constraints and align the child based on
+/// [alignment].  If the container cannot expand enough to accommodate the
+/// entire child, the child will be clipped.
+///
+/// In debug mode, if the child overflows the container, a warning will be
+/// printed on the console, and black and yellow striped areas will appear where
+/// the overflow occurs.
+///
+/// See also:
+///
+///  * [ConstrainedBox] for a box which imposes constraints on its child.
+///  * [Container], a convenience widget that combines common painting,
+///    positioning, and sizing widgets.
+///  * [OverflowBox], a widget that imposes different constraints on its child
+///    than it gets from its parent, possibly allowing the child to overflow
+///    the parent.
+///  * [RenderUnconstrainedBox] for the [RenderObject] for this container.
+///  * The [catalog of layout containers](https://flutter.io/widgets/layout/).
+class RenderUnconstrainedBox extends RenderBox
+  with RenderObjectWithChildMixin<RenderBox>,
+       RenderProxyBoxMixin,
+       DebugOverflowIndicatorMixin {
+  RenderUnconstrainedBox({
+    RenderBox child,
+    @required TextDirection textDirection,
+    @required AlignmentGeometry alignment,
+  }) : assert(alignment != null),
+       _textDirection = textDirection,
+       _alignment = alignment {
+    this.child = child;
+  }
+
+  /// Determines the order to lay children out horizontally and how to interpret
+  /// `start` and `end` in the horizontal direction for the alignment.
+  ///
+  /// The textDirection is only used when [alignment] is an
+  /// [AlignmentDirectional], but must be non-null in that case.
+  @override
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection != value) {
+      _textDirection = value;
+      markNeedsLayout();
+    }
+  }
+
+  /// How to align the the child in the box.
+  ///
+  /// If this is set to an [AlignmentDirectional] object, then [textDirection]
+  /// must not be null.
+  ///
+  /// See also:
+  ///  * [AlignmentDirectional] for direction-aware alignment.
+  ///  * [Alignment] for non-direction-aware alignment.
+  AlignmentGeometry get alignment => _alignment;
+  AlignmentGeometry _alignment;
+  set alignment(AlignmentGeometry value) {
+    assert(value != null);
+    if (_alignment == value)
+      return;
+    _alignment = value;
+    _markNeedResolution();
+  }
+
+  Alignment _resolvedAlignment;
+  void _resolve() {
+    _resolvedAlignment ??= alignment.resolve(textDirection ?? TextDirection.ltr);
+  }
+
+  void _markNeedResolution() {
+    _resolvedAlignment = null;
+    markNeedsLayout();
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! BoxParentData)
+      child.parentData = new BoxParentData();
+  }
```

Comment:
if you're using parentData, you definitely don't want to be a proxy box.

Reply:
OK.

Reply:
i think you can drop this now that your superclass takes care of it